### PR TITLE
Bug fix in signal desktop update script

### PR DESCRIPTION
**Pull request by Arjun31415**
## Description of changes

fix `jq` not present in nix-shell.
fix `nix-update` command arguments

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Arjun31415** [comment]
\cc @Mic92

**Comment by Arjun31415** [comment]
I would also like to let you know that the update script does not work for me in case of `signal-desktop-beta`
```
export UPDATE_NIX_ATTR_PATH='signal-desktop-beta' && ./pkgs/applications/networking/instant-messengers/signal-desktop/update.sh
``` 

```
+++ dirname ./pkgs/applications/networking/instant-messengers/signal-desktop/update.sh
++ cd ./pkgs/applications/networking/instant-messengers/signal-desktop
++ pwd
+ SCRIPT_DIR=/home/prometheus/Nixpkgs/nixpkgs/pkgs/applications/networking/instant-messengers/signal-desktop
+ case "$UPDATE_NIX_ATTR_PATH" in
++ curl_github https://api.github.com/repos/signalapp/Signal-Desktop/releases
++ curl https://api.github.com/repos/signalapp/Signal-Desktop/releases
++ jq -r '.[0].tag_name'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 81398    0 81398    0     0  77762      0 --:--:--  0:00:01 --:--:-- 77818
+ latestTagBeta=v6.42.0-beta.1
++ expr v6.42.0-beta.1 : 'v\(.*\)'
+ latestVersionBeta=6.42.0-beta.1
+ echo 'Updating signal-desktop-beta for x86_64-linux'
Updating signal-desktop-beta for x86_64-linux
+ nix-update --version 6.42.0-beta.1 --system x86_64-linux --override-filename /home/prometheus/Nixpkgs/nixpkgs/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-beta.nix signal-desktop-beta
$ nix eval --json --impure --expr 
let
  
  pkgs = import "/home/prometheus/Nixpkgs/nixpkgs";
  args =  builtins.functionArgs pkgs;
  inputs = (if args ? system then { system = "x86_64-linux"; } else {}) //
           (if args ? overlays then { overlays = [ ]; } else {});
  pkg = (pkgs inputs)."signal-desktop-beta";
  sanitizePosition = x: x;

  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);

  position = if pkg ? isRubyGem then
    raw_version_position
  else if pkg ? isPhpExtension then
    raw_version_position
   else
    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
in {
  name = pkg.name;
  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;
  inherit raw_version_position;
  filename = position.file;
  line = position.line;
  urls = pkg.src.urls or null;
  url = pkg.src.url or null;
  rev = pkg.src.rev or null;
  hash = pkg.src.outputHash or null;
  go_modules = pkg.goModules.outputHash or null;
  go_modules_old = pkg.go-modules.outputHash or null;
  cargo_deps = pkg.cargoDeps.outputHash or null;
  raw_cargo_lock =
    if pkg ? cargoDeps.lockFile then
      let
        inherit (pkg.cargoDeps) lockFile;
        res = builtins.tryEval (sanitizePosition {
          file = toString lockFile;
        });
      in
      if res.success then res.value.file else false
    else
      null;
  npm_deps = pkg.npmDeps.outputHash or null;
  yarn_deps = pkg.offlineCache.outputHash or null;
  tests = builtins.attrNames (pkg.passthru.tests or {});
  has_update_script = pkg.passthru.updateScript or null != null;
  src_homepage = pkg.src.meta.homepage or null;
  changelog = pkg.meta.changelog or null;
} --system x86_64-linux --extra-experimental-features flakes nix-command
error:
       … while evaluating attribute 'filename'

         at «string»:23:3:

           22|   inherit raw_version_position;
           23|   filename = position.file;
             |   ^
           24|   line = position.line;

       … while calling the 'unsafeGetAttrPos' builtin

         at «string»:18:23:

           17|    else
           18|     sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
             |                       ^
           19| in {

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: value is a path while a set was expected
Traceback (most recent call last):
  File "/nix/store/rss9z6n3rki1xhzlzgq47c7z5qwq3ikx-nix-update-1.0.0/bin/.nix-update-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/rss9z6n3rki1xhzlzgq47c7z5qwq3ikx-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/__init__.py", line 287, in main
    package = update(options)
              ^^^^^^^^^^^^^^^
  File "/nix/store/rss9z6n3rki1xhzlzgq47c7z5qwq3ikx-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/update.py", line 322, in update
    package = eval_attr(opts)
              ^^^^^^^^^^^^^^^
  File "/nix/store/rss9z6n3rki1xhzlzgq47c7z5qwq3ikx-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/eval.py", line 181, in eval_attr
    res = run(cmd)
          ^^^^^^^^
  File "/nix/store/rss9z6n3rki1xhzlzgq47c7z5qwq3ikx-nix-update-1.0.0/lib/python3.11/site-packages/nix_update/utils.py", line 37, in run
    return subprocess.run(
           ^^^^^^^^^^^^^^^
  File "/nix/store/qp5zys77biz7imbk6yy85q5pdv7qk84j-python3-3.11.6/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nix', 'eval', '--json', '--impure', '--expr', '\nlet\n  \n  pkgs = import "/home/prometheus/Nixpkgs/nixpkgs";\n  args =  builtins.functionArgs pkgs;\n  inputs = (if args ? system then { system = "x86_64-linux"; } else {}) //\n           (if args ? overlays then { overlays = [ ]; } else {});\n  pkg = (pkgs inputs)."signal-desktop-beta";\n  sanitizePosition = x: x;\n\n  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);\n\n  position = if pkg ? isRubyGem then\n    raw_version_position\n  else if pkg ? isPhpExtension then\n    raw_version_position\n   else\n    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);\nin {\n  name = pkg.name;\n  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;\n  inherit raw_version_position;\n  filename = position.file;\n  line = position.line;\n  urls = pkg.src.urls or null;\n  url = pkg.src.url or null;\n  rev = pkg.src.rev or null;\n  hash = pkg.src.outputHash or null;\n  go_modules = pkg.goModules.outputHash or null;\n  go_modules_old = pkg.go-modules.outputHash or null;\n  cargo_deps = pkg.cargoDeps.outputHash or null;\n  raw_cargo_lock =\n    if pkg ? cargoDeps.lockFile then\n      let\n        inherit (pkg.cargoDeps) lockFile;\n        res = builtins.tryEval (sanitizePosition {\n          file = toString lockFile;\n        });\n      in\n      if res.success then res.value.file else false\n    else\n      null;\n  npm_deps = pkg.npmDeps.outputHash or null;\n  yarn_deps = pkg.offlineCache.outputHash or null;\n  tests = builtins.attrNames (pkg.passthru.tests or {});\n  has_update_script = pkg.passthru.updateScript or null != null;\n  src_homepage = pkg.src.meta.homepage or null;\n  changelog = pkg.meta.changelog or null;\n}', '--system', 'x86_64-linux', '--extra-experimental-features', 'flakes nix-command']' returned non-zero exit status 1.
```

**Review by teutat3s — APPROVED** [review]

**Review by teutat3s — COMMENTED** [review]
Please change the commit message to something like:

```
signal-desktop: fix jq not present in nix-shell
fix `nix-update` command arguments
```

**Comment by Arjun31415** [comment]
```
 nix-build -A signal-desktop-beta
error: expression does not evaluate to a derivation (or a set or list of those)
```
For some reason signal-desktop-beta is not recognized also

**Comment by Arjun31415** [comment]
the output of the script after these commits - 

```patch
diff --git a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-aarch64.nix b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-aarch64.nix
index d69571928af8..a38878bbac64 100644
--- a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-aarch64.nix
+++ b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-aarch64.nix
@@ -2,7 +2,7 @@
 callPackage ./generic.nix { } rec {
   pname = "signal-desktop";
   dir = "Signal";
-  version = "6.40.0";
+  version = "6.41.0";
   url = "https://github.com/0mniteck/Signal-Desktop-Mobian/raw/${version}/builds/release/signal-desktop_${version}_arm64.deb";
   hash = "sha256-3Pi0c+CGcJR1M4ll51m+B5PmGIcIjjlc0qa9b8rkMeU=";
 }
diff --git a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-beta.nix b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-beta.nix
index 9b99f67e156a..e61e3f6fd76e 100644
--- a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-beta.nix
+++ b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop-beta.nix
@@ -2,7 +2,7 @@
 callPackage ./generic.nix {} rec {
   pname = "signal-desktop-beta";
   dir = "Signal Beta";
-  version = "6.40.0-beta.2";
+  version = "6.42.0-beta.1";
   url = "https://updates.signal.org/desktop/apt/pool/s/signal-desktop-beta/signal-desktop-beta_${version}_amd64.deb";
-  hash = "sha256-pfedkxbZ25DFgz+/N7ZEb9LwKrHuoMM+Zi+Tc21QPsg=";
+  hash = "sha256-H+7OvosWQSfEr5nXIba3ArtSgCGkUz4XLG0LkfOcudE=";
 }
diff --git a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop.nix b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop.nix
index bfae729772f4..8f1ac18e45ab 100644
--- a/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop.nix
+++ b/pkgs/applications/networking/instant-messengers/signal-desktop/signal-desktop.nix
@@ -2,7 +2,7 @@
 callPackage ./generic.nix {} rec {
   pname = "signal-desktop";
   dir = "Signal";
-  version = "6.40.0";
+  version = "6.41.0";
   url = "https://updates.signal.org/desktop/apt/pool/s/signal-desktop/signal-desktop_${version}_amd64.deb";
-  hash = "sha256-vyXHlycPSyEyv938IKzGM6pdERHHerx2CLY/U+WMrH4=";
+  hash = "sha256-DwlynyMjUVcyM6K1lSUTmG1Vgw3juJFCrVZnJa5qyXg=";
 }
```

File: pkgs/applications/networking/instant-messengers/signal-desktop/default.nix [hunk 1]
```diff
@@ -2,5 +2,5 @@
   signal-desktop = if hostPlatform.system == "aarch64-linux"
     then callPackage ./signal-desktop-aarch64.nix { }
     else callPackage ./signal-desktop.nix { };
-  signal-desktop-beta = ./signal-desktop-beta.nix;
+  signal-desktop-beta = callPackage ./signal-desktop-beta.nix{ };
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/signal-desktop/default.nix:5]: Interesting how this even evaluated...

**Pull request merged**